tests/Unit/Service/Phishing/DateCheckTest.php

```diff
@@ -65,4 +65,23 @@ public function testInTheFuture(): void {
 		$this->assertTrue($result->isPhishing());
 	}
 
+	public function testInvalidDate(): void {
+
+		$this->time->expects($this->exactly(2))
+			->method('getDateTime')
+			->willReturnCallback(function ($argument): DateTime {
+				static $callCount = 0;
+				$callCount++;
+
+				if ($callCount === 1) {
+					return new \DateTime('now');
+				}
+				throw new \DateException();
```

Comment:
use `match` of `if` on the argument. then you make the mock dependent on the input, not the order of calls. This is also the reason why the consecutive method in phpunit was removed.